src/Mvc/Mvc.Core/src/ChallengeResult.cs

```diff
@@ -97,7 +98,18 @@ public override async Task ExecuteResultAsync(ActionContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            await ExecuteAsync(context.HttpContext);
+        }
+
+        /// <inheritdoc />
+        async Task IResult.ExecuteAsync(HttpContext httpContext)
+        {
+            await ExecuteAsync(httpContext);
```

Comment:
Remove the async/await, just pass through

Reply:
Done.